internal/integration/parse.go

```diff
@@ -61,7 +61,11 @@ func readResult(r io.Reader) (map[string]string, string, [][]byte, error) {
 	// Read all of the output so that we can include it with errors.
 	data, err := io.ReadAll(r)
 	if err != nil {
-		return nil, "", nil, err
+		if len(data) > 0 {
+			t.Logf("got data, but there was an error: %v", err)
+		} else {
+			return nil, "", nil, err
+		}
```

Comment:
This change works around a similar problem to the one described in https://github.com/hashicorp/go-plugin/issues/116, where there's a race between reads from a pipe and `exec.Cmd.Wait` closing the pipe after the process exits.